### DISCUSSION TITLE
Check if feature intersects area (OSM data)

### DISCUSSION THREAD

**Discussion by michuapps** [systemed/tilemaker#704] [Q&A]
Hi,

Can we check if the feature intersects with specific area that's in the same .osm.pbf file? I see there are functions like Intersects, FindIntersecting,, CoveredBy, FindCovering, but they only work with provided additional shapefile layers. I'am looking for this kind of function but to check inside .osm.pbf.

So my problem is that I need to Layer() all the features that are inside specific areas. e.g. I want all buildings or roads that are somehow touching leisure=park or amenity=school. Is it possible?

**Comment by systemed** [comment]
It's not possible with tilemaker at present and I don't think it would be particularly plausible to implement, I'm afraid.

Shapefiles are loaded before OSM data. That means when we're running `way_function`, all the shapefile data is in memory for us to query against. That isn't the case for other OSM data - it might or might not have been loaded yet.
